CleanWeather/Scenes/CityList/CityListWorker.swift

```diff
@@ -30,23 +32,59 @@ extension CityListWorkerImpl: CityListWorker {
         repository.fetchFavouriteCities(completion: completion)
     }
 
-    func fetchCityWeather(completion: FetchWeatherCompletion?) {
-        networking.fetchCurrentWeatherForAllCities { [weak self] result in
-            
-            guard let self = self, case .success(let city) = result else {
-                completion?(result)
-                return
+    func fetchCitiesWeather(cities: [City], completion: FetchWeatherCompletion?) {
+       var responses = [Result<CityWeather, Error>]()
+        cities.forEach { city in
+            let coordinates = Coordinates(lat: city.latitude, lon: city.longitude)
+            networking.fetchCurrentWeatherForCity(coordinates: coordinates.stringValue) { response in
+                let cityWeatherResult = self.convertToCityWeatherResult(city: city, response: response)
+                responses.append(cityWeatherResult)
+                self.handleCityWeatherResponse(cities: cities, responses: responses, completion: completion)
             }
-            
-            let sortedCity = self.sortCity(city: city)
-            completion?(.success(sortedCity))
         }
     }
 }
 
 private extension CityListWorkerImpl {
-    
-    private func sortCity(city: [CityWeather]) -> [CityWeather] {
-        return city.sorted(by: { $0.city < $1.city })
+
+    private func handleCityWeatherResponse(cities: [City], responses: [Result<CityWeather, Error>], completion: FetchWeatherCompletion?) {
+        guard cities.count == responses.count else { return }
+        let citiesWeatherListResult = convertToCityWeatherListResult(responses: responses)
+        completion?(citiesWeatherListResult)
+    }
+
+    private func convertToCityWeatherResult(city: City, response: Result<CityListAPIResponse, Error>) -> Result<CityWeather, Error> {
+        switch response {
```

Comment:
did you try these ENUM-mapping?
just thinking about it :)

Reply:
Enum mapping?
Could you write more about that?